### ISSUE TITLE
ID4178 when setting EncryptingCertificate on RelyingParty

### ISSUE THREAD

**Issue by vidarkongsli** [IdentityServer/IdentityServer3.WsFederation#30]
I am trying to have IdSrv encrypt the assertion sent to a certain relying party. I load a .cer file with the certificate of the RP, and set it on the relying party definition:
```csharp
new RelyingParty
{
    Realm = "urn:client",
    Name = "Client name",
    Enabled = true,
    ReplyUrl = "https://localhost:44303/wtrealm",
    TokenType = TokenTypes.Saml2TokenProfile11,
    TokenLifeTime = 30,
    IncludeAllClaimsForUser = true,
    EncryptingCertificate = Certificate.LoadClientCertFromCerFile() // X509Certificate2
}
```
I then get the following exception when IdSrv is trying to issue the assertion:

    ID4178: The EncryptingCredentials provided in the SecurityTokenDescriptor is for an asymmetric key. You must use an EncryptedKeyEncryptingCredentials to encrypt the token.

**Comment by vidarkongsli** [comment]
It blows up on the following line inside `SignInResponseGenerator`:
```csharp
descriptor.EncryptingCredentials
   = new X509EncryptingCredentials(validationResult.RelyingParty.EncryptingCertificate);
```
Should the framework not create a ```new EncryptedKeyEncryptingCredentials(
validationResult.RelyingParty.EncryptingCertificate)``` instead?

**Comment by leastprivilege** [comment]
You are probably right - wanna send a PR? Read the the contribution guidelines first please. thanks!

**Comment by vidarkongsli** [comment]
I'll have a look

**Comment by vidarkongsli** [comment]
Created a PR. I have tested it with an MVC .NET 4.5 with IdentityModel configuration as client.

**Comment by leastprivilege** [comment]
done

**Comment by WhiteBone** [comment]
I still blows up.
Is there a reason why the fix was reverted? 

https://github.com/IdentityServer/IdentityServer3.WsFederation/commit/6b4a7363a819b492dd75e2b74f6b99b67b5da28a

**Comment by vidarkongsli** [comment]
@WhiteBone The fix was reapplied here https://github.com/IdentityServer/IdentityServer3.WsFederation/commit/a290ed0952f1ef13c0a04ffb48c785f671a2086e

**Comment by WhiteBone** [comment]
@vidarkongsli I might be slightly blind. But as I see it it was fixed in https://github.com/IdentityServer/IdentityServer3.WsFederation/commit/a290ed0952f1ef13c0a04ffb48c785f671a2086e and reverted 17 minutes later in https://github.com/IdentityServer/IdentityServer3.WsFederation/commit/6b4a7363a819b492dd75e2b74f6b99b67b5da28a.

**Comment by mattrandle** [comment]
Yes the latest copy of the source still has 

descriptor.EncryptingCredentials
   = new X509EncryptingCredentials(validationResult.RelyingParty.EncryptingCertificate);

Instead of ,

new EncryptedKeyEncryptingCredentials( validationResult.RelyingParty.EncryptingCertificate)

Is there a reason why the fix was backed out or does it a merge issue and just needs reapplying?

Without this, you cant encrypt ws-fed responses.

**Comment by brockallen** [comment]
Not sure what happened. I'll reopen to track.

**Comment by Thor-Kristian-Valderhaug** [comment]
I also need this fix to be able to encrypt ws-fed responses.
Do we have ETA on when this is fixed ?

**Comment by vidarkongsli** [comment]
Created a PR that reapplies the fix in the DEV branch

**Comment by brockallen** [comment]
Fixed in 2.0.1
